client/scripts/helpers/search.ts

```diff
@@ -2,6 +2,33 @@ import $ from 'jquery';
 import app from 'state';
 import m from 'mithril';
 import { SearchParams } from '../views/components/search_bar';
+import { modelFromServer } from '../controllers/server/threads';
+import { OffchainThread } from '../models';
+
+export const searchThreadTitles = async (
+  searchTerm: string,
+  params: SearchParams
+): Promise<OffchainThread[]> => {
+  const { resultSize, chainScope, communityScope } = params;
+  try {
```

Comment:
Is this `try/catch` necessary if you have a `if (response.status !== 'Success') {` further below? 

I may be wrong but I believe when `$.get(...)` receives a 40X/50X error, it'll throw instead of hitting that `if` statement, and thus the `if` is instead not necessary.

Reply:
You're right, but the `modelFromServer` function is liable to throwing. That said, we could change the structure of try/catches across the codebase, to avoid redundancy—it's just that this is a pretty standard pattern around our controllers, see also threads, comments, etc. That said, if you wanted to refactor codebase-wide, it'd be a welcome shift!

Reply:
(Unfortunately) we have this pattern all over the place as @gdjohnson notes. We can talk about fixing the redundancy of this style choice soon, but yes, we sometimes have data formatting issues and the try/catch is the implementation decision that was made.